gui/middleware/notifier.py

```diff
@@ -812,6 +812,8 @@ def mp_change_permission(self, path='/mnt', user=None, group=None,
         winacl = os.path.join(path, ACL_WINDOWS_FILE)
         macacl = os.path.join(path, ACL_MAC_FILE)
         winexists = (ACL.get_acl_ostype(path) == ACL_FLAGS_OS_WINDOWS)
+        zfs_dataset = libzfs.ZFS().get_dataset_by_path(path)
```

Comment:
libzfs is not thread-safe and this can run in a thread.

Workaround for it is using context manager:

```
with libzfs.ZFS() as zfs:
    zfs_dataset_name = zfs.get_dataset_by_path(path).name
```